arelle/ValidateXbrlDTS.py

```diff
@@ -262,9 +262,10 @@ def checkDTS(val: ValidateXbrl, modelDocument: ModelDocument.ModelDocument, chec
     val.extendedElementName = None
     isFilingDocument = False
     # validate contents of entry point document or its sibling/descendant documents or in report package of entry point
-    if ((modelDocument.uri.startswith(val.modelXbrl.uriDir) or # document uri in same subtree as entry doocument
-         (val.modelXbrl.fileSource.isOpen and modelDocument.filepath.startswith(val.modelXbrl.fileSource.baseurl))) and # document in entry submission's package
-        modelDocument.targetNamespace not in val.disclosureSystem.baseTaxonomyNamespaces and
+    if ((((modelDocument.uri.startswith(val.modelXbrl.uriDir) or # document uri in same subtree as entry doocument
+          (val.modelXbrl.fileSource.isOpen and modelDocument.filepath.startswith(val.modelXbrl.fileSource.baseurl))) and # document in entry submission's package
+          modelDocument.targetNamespace not in val.disclosureSystem.baseTaxonomyNamespaces)
+         or not val.modelXbrl.modelManager.skipBaseTaxonomiesValidation) and
```

Comment:
There's a lot going on in this one conditional. It looks like there's a bug in how disclosure system base taxonomy namespaces are handled, but it's hard to reason about with how this is written. I've talked with two other devs about these lines and we all came to separate conclusions about what it was doing. Can we clean this up instead of introducing another case inline? Perhaps break it out into a function?

```python
def shouldValidateDocumentContents(val, modelDocument):
    if modelDocument.xmlDocument is None:
        return False

    if modelDocument.targetNamespace in val.disclosureSystem.baseTaxonomyNamespaces:
        # Plugin has declared this as a known good namespace. Skip validation.
        return False

    if not val.modelXbrl.modelManager.skipBaseTaxonomiesValidation:
        return True

    if modelDocument.uri.startswith(val.modelXbrl.uriDir):
        return True

    return val.modelXbrl.fileSource.isOpen and modelDocument.filepath.startswith(val.modelXbrl.fileSource.baseurl)
```